### PR TITLE
change Solaar config file from json to yaml

### PR DESCRIPTION
**Pull request by pfps**
Json has limitations which makes Solaar code more complex.  Using YAML allows for better code in the rest of Solaar.

### CODE REVIEW

**Comment by pfps** [comment]
@barolo @sk8geek @mschilli87 @viniciusbm  Can you test this PR?

Test by running `bin/solaar -ddd`.  Check  for errors and warnings.  Then check whether all expected settings show up.  Change some settings, including boolean settings, ranged settings, and complex settings.  Quit out of Solaar.  Then check ~/.config/solaar/config.yaml to see if it looks reasonable.  Start Solaar again and again check for errors and warnings and expected settings.

You can then go back to your regular version of Solaar.  The old config file will still be available.

**Comment by viniciusbm** [comment]
There seems to be something wrong with "Key/Button Diversion": 

image

[log1.txt](https://github.com/pwr-Solaar/Solaar/files/8438877/log1.txt)

Perhaps it's related to type conversion, because it's saving the boolean values for this setting as integer numbers (but not for the other settings): 

```yaml
  disable-keyboard-keys: {1: false, 4: false, 8: false, 16: false}
  divert-keys: {1: 0, 2: 0, 3: 0, 4: 1, 5: 1, 6: 1, 8: 1, 34: 1, 40: 1, 62: 1, 64: 0, 79: 1, 84: 1, 110: 1, 145: 1, 149: 1, 183: 1, 184: 1}
```


I'll check the other things.

**Comment by viniciusbm** [comment]
> Change some settings, including boolean settings, ranged settings, and complex settings. Quit out of Solaar. Then check ~/.config/solaar/config.yaml to see if it looks reasonable. Start Solaar again and again check for errors and warnings and expected settings.

Everything seems to be working as expected, except for the UI glitch I mentioned above.

File: lib/logitech_receiver/settings.py [hunk 1]
```diff
@@ -472,7 +472,7 @@ def read(self, cached=True):
                 if reply:
                     # keys are ints, because that is what the device uses,
                     # encoded into strings because JSON requires strings as keys
```
Comment (viniciusbm) [inline lib/logitech_receiver/settings.py:474]: This comment should probably be removed

File: lib/logitech_receiver/settings.py [hunk 2]
```diff
@@ -371,7 +371,7 @@ def read(self, cached=True):
...
```
Comment (viniciusbm) [inline lib/logitech_receiver/settings.py:373]: This comment should probably be removed

**Comment by pfps** [comment]
@viniciusbm Good catch on the UI.  That was from a fairly serious bug.

File: lib/logitech_receiver/settings.py (hunk 1 above)
Comment (pfps) [reply lib/logitech_receiver/settings.py:474]: Done

File: lib/logitech_receiver/settings.py (hunk 2 above)
Comment (pfps) [reply lib/logitech_receiver/settings.py:373]: Done

**Comment by pfps** [comment]
The bug should be fixed.  Retesting would be useful.

**Comment by viniciusbm** [comment]
> The bug should be fixed. Retesting would be useful.

I've just tested the latest version (466856d44486ed7a80020eda20bd8329e55fd93d), and everything seems to be working. 

The program reads the existing JSON file and saves a YAML file correctly, without errors or warnings. I changed at least one item for each setting my device supports and checked that the behaviour changes as expected and the new value is stored correctly. 

[log3.txt](https://github.com/pwr-Solaar/Solaar/files/8443623/log3.txt)

Then, I closed Solaar, opened it again and checked that the new values are still there. 

I also ran it without any configuration file, and it produces a new YAML file as expected.

**Pull request merged**